TLM/TLM/Manager/Impl/LaneArrowManager.cs

```diff
@@ -117,19 +117,16 @@ public class LaneArrowManager
         /// <summary>
         /// Resets lane arrows to their default value for the given segment end.
         /// </summary>
-        /// <param name="segmentId">segment to reset</param>
-        /// <param name="startNode">determines the segment end to reset. if <c>null</c>
-        /// both ends are reset</param>
+        /// <param name="segmentId">Segment id.</param>
+        /// <param name="startNode">Determines the segment end to reset, or both if <c>null</c>.</param>
         public void ResetLaneArrows(ushort segmentId, bool? startNode = null) {
-            ExtSegmentManager extSegmentManager = ExtSegmentManager.Instance;
-            foreach (var lane in extSegmentManager.GetSortedLanes(
-                segmentId,
-                ref segmentId.ToSegment(),
-                startNode,
-                LANE_TYPES,
-                VEHICLE_TYPES)) {
+
+            ref NetSegment segment = ref segmentId.ToSegment();
+
+            var sortedLanes = segment.GetSortedLanes(startNode, LANE_TYPES, VEHICLE_TYPES);
```

Comment:
I could probably add `sort: false` here?

Reply:
its not performance critical part of the code but it would be a minor improvement.
maybe we can have another overload of `segment.GetLaneIndexAndLaneIds()` that can filter by lane/vehicle types. 
But I don't recall usage of this method in performance critical part of the code so who cares! maybe we should document that it is slow. is there an annotation for that?

Reply:
> is there an annotation for that?

Not that I'm aware of. I was hunting round a couple of weeks back to see if there were any annotations for defining hotpath, etc., and couldn't find anything.

Reply:
updated to skip sorting